File: leetcode/643.maximum-average-subarray-i.cpp
/*
 * @lc app=leetcode id=643 lang=cpp
 *
 * [643] Maximum Average Subarray I
 */

// @lc code=start
class Solution {
public:

	double findMaxAverage(vector<int>& nums, int k) {
		int		i = 0;
		int		len = nums.size();
		double	sum = 0;

		while (i < k) {
			sum += nums[i];
			i++;
		}
		double max_sum = sum;
		while (i < len) {
			sum = sum - nums[i - k] + nums[i];
			max_sum = max(sum, max_sum);
			i++;
		}
		return max_sum / k;
	}
};
// @lc code=end

File: leetcode/643.maximum-average-subarray-i.py
#
# @lc app=leetcode id=643 lang=python3
#
# [643] Maximum Average Subarray I
#

# @lc code=start
class Solution:
	def findMaxAverage(self, nums: List[int], k: int) -> float:
		sum, maxSum = 0, 0
		for i in range(len(nums)):
			if i < k:
				sum += nums[i]
				maxSum += nums[i]
			else:
				sum += nums[i] - nums[i - k]
				maxSum = max(sum, maxSum)
		return maxSum / k
# @lc code=end

